BlueprintUI/Sources/Layout/Flow.swift

```diff
@@ -0,0 +1,390 @@
+import UIKit
+
+
+/// Element which lays out children horizontally, wrapping to another row when there is not enough space.
+///
+/// You may control the layout of each row within the flow layout, by providing a `HorizontalAlignment`,
+/// which controls the horizontal alignment of content within a row when it is smaller than its container,
+/// and via `RowAlignment`, which controls the vertical alignment of shorter elements within a row.
+///
+/// ```swift
+/// Flow(horizontalSpacing: 10, rowSpacing: 10) {
+///     OnboardingPill("Food / drink")
+///     OnboardingPill("Retail goods").flowChild(key: "aKey")
+///     OnboardingPill("Grocery / gourmet / alcohol")
+///     OnboardingPill("Beauty / wellness bookings")
+///     OnboardingPill("Healthcare services")
+///     OnboardingPill("Something else")
+/// }
+/// ```
+///
+/// Below is a diagram showing a simple example of a `Flow` layout.
+///
+/// ```
+/// ┌─────────────────────────────────────┐
+/// │ ┌─────────┐┌─────────┐┌────────────┐│
+/// │ │    1    ││    2    ││     3      ││
+/// │ └─────────┘└─────────┘└────────────┘│
+/// │ ┌───────┐┌─────┐┌───────────┐       │
+/// │ │   4   ││  5  ││     6     │       │
+/// │ └───────┘└─────┘└───────────┘       │
+/// └─────────────────────────────────────┘
+/// ```
+public struct Flow: Element {
```

Comment:
Was wondering the same myself if we ever needed to build a vertical flow (similar to Google Image search). We could use the term `Waterfall` for that one ([example](https://github.com/sgr-ksmt/WaterfallLayout)) and stick with `Flow` for the horizontal one.

Reply:
I'd figure if we ever want this, we could add a `direction` parameter, like we have for `EqualStack` – thoughts?

Reply:
The implementations of the two orientations and the exposed APIs would be different enough that it might not make a lot of sense to have one struct be used for both? Don't have a strong opinion on this 🤷

Reply:
Yeah, it seems like working with the configuration properties like `horizontalAlignment`, `horizontalSpacing` etc would be confusing to work with in a vertical `direction`.

If we're thinking this would evolve into supporting vertical direction then it might make sense to use more generic property names, like `UICollectionViewFlowLayout` does (`minimumLineSpacing`, `minimumInteritemSpacing`, etc).